lib/StripeResource.js

```diff
@@ -378,7 +392,12 @@ StripeResource.prototype = {
     };
 
     const makeRequest = (apiVersion, headers, numRetries) => {
-      const timeout = this._stripe.getApiField('timeout');
+      // timeout can be set on a per-request basis. Favor that over the global setting
+      const timeout =
+        options.settings && options.settings.timeout
```

Comment:
I think we want `options.settings && Number.isInteger(options.settings.timeout)` so people can set it to zero, no?

🤔 not actually sure a timeout of zero makes any sense though... Users might want it for some kind of test situation?

Reply:
== It might make sense for this to be a positive integer, and maybe > 0?

Reply:
I'm going to add 
```javascript
Number.isInteger(options.settings.timeout) && options.settings.timeout >= 0
```
As for test purposes a timeout of 0 could make sense.